packages/components/resource-pages/ResourcePopOver.tsx

```diff
@@ -29,16 +39,57 @@ const ResourcePopOver: React.FC<ResourcePopOverProps> = ({
   );
 };
 
+export const BucketClassPopOver: React.FC<BucketClassPopOverProps> = ({
+  label,
+  bucketClasses,
+  title,
+}) => {
+  const { t } = useTranslation();
+
+  return (
+    <ResourcePopOver
+      label={label}
+      title={title ? title : t('Connected data sources')}
+    >
+      <div className="resource-pop-over">
+        {(bucketClasses?.length > MAX_NO_OF_RESOURCE_DISPLAY
+          ? bucketClasses.slice(0, MAX_NO_OF_RESOURCE_DISPLAY)
+          : bucketClasses
+        )?.map((resourceName) => (
+          <ResourceLink
+            link={`/mcgms/resource/${referenceForModel(
+              NooBaaBucketClassModel
+            )}/${resourceName}`}
+            resourceName={resourceName}
+            key={resourceName}
+            hideIcon
+          />
+        ))}
+      </div>
+      {bucketClasses.length > MAX_NO_OF_RESOURCE_DISPLAY && (
+        <Link to={BUCKET_CLASS_LIST_PATH}>{t('view more')}</Link>
+      )}
+    </ResourcePopOver>
+  );
+};
+
 export const DataResourcesPopOver: React.FC<DataResourcesPopOverProps> = ({
   label,
   dataResources,
+  title,
 }) => {
   const { t } = useTranslation();
 
   return (
-    <ResourcePopOver label={label} title={t('Connected data sources')}>
+    <ResourcePopOver
+      label={label}
+      title={title ? title : t('Connected data sources')}
+    >
       <div className="resource-pop-over">
-        {dataResources?.map((resourceName) => (
+        {(dataResources?.length > MAX_NO_OF_RESOURCE_DISPLAY
+          ? dataResources.slice(0, MAX_NO_OF_RESOURCE_DISPLAY)
+          : dataResources
+        )?.map((resourceName) => (
```

Comment:
u can't make this change here... this component is used in list page also...
there we want to show all the items... not just 5...

Reply:
or if your r... try to make the component more generic...

Reply:
[comment](https://github.com/red-hat-storage/mcg-ms-console/pull/21#discussion_r891466261)